csrc/device_lower/utils.h

```diff
@@ -304,6 +304,10 @@ bool isExtentEqualToMaxParallelTypeExtent(const IterDomain* id);
 //! indexing special items in shared memory, like mbarrier.
 NVF_API Val* u32IndexScalarSmemTv(TensorView* tv);
 
+//! Get the uint32_t index of a TensorIndex. This is usually used for
```

Comment:
nit: I guess you copy-and-pasted the function above, but why is it called "scalar smem TV"? It seems we already has an address as a `Val`, so it doesn't seem to matter if it's scalar or not.

Reply:
A single `mbarrier` and its token are represented as a size-0 `int64_t` TensorView in shared memory, which can be called a scalar, smem TV.